Dalamud/Game/Gui/ContextMenu/CharacterData.cs

```diff
@@ -0,0 +1,472 @@
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Dalamud.Memory;
+using FFXIVClientStructs.FFXIV.Client.UI.Info;
+
+namespace Dalamud.Game.Gui.ContextMenu;
+
+/// <summary>
+/// Dalamud wrapper around a ClientStructs CharacterData.
+/// </summary>
+[StructLayout(LayoutKind.Explicit, Size = StructSizeInBytes)]
+public unsafe struct CharacterData
+{
+    /// <summary>
+    /// The actual data.
+    /// </summary>
+    [FieldOffset(0)]
+    internal readonly InfoProxyCommonList.CharacterData InternalData;
+
+    private const int StructSizeInBytes = 0x68;
+
+    /// <summary>
+    /// The view of the backing data, in <see cref="ulong"/>.
+    /// </summary>
+    [FieldOffset(0)]
+    private fixed ulong dataUInt64[StructSizeInBytes / 0x8];
```

Comment:
For `GameInventoryItem` struct, this was used to make `Equals` and `GetHashCode` fast; is that purpose also applicable here?